pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -176,18 +179,26 @@ func (c *Controller) Apply(change *sink.Change) error {
 	}
 
 	var prevStore map[string]map[string]*model.Config
-
 	c.configStoreMu.Lock()
 	prevStore = c.configStore[descriptor.Type]
 	c.configStore[descriptor.Type] = innerStore
 	c.configStoreMu.Unlock()
 
-	if descriptor.Type == model.ServiceEntry.Type {
+	if c.options.IncrementalEDS && descriptor.Collection == model.SyntheticServiceEntry.Collection {
+		if err := c.edsUpdate(innerStore, prevStore); err != nil {
```

Comment:
Looking at `edsUpdate` it appears to do a comparison between the previous resources and the new.  Is this needed in the case of incremental? With incremental, all of the resources that we're receiving are updated ... we just need to determine what was updated: service, endpoints, or both.

If any of them had an update to the service part, then we have to do a *full update* (e.g. `c.options.XDSUpdater.ConfigUpdate(true)`). Otherwise, we can do a one-by-one EDS update.

Reply:
I will update as per our discussion.